### PR TITLE
Added config option to disable teleporting nearby animals

### PR DESCRIPTION
**Pull request by bittiez**
Also fixed a few deprecated methods

### CODE REVIEW

**Comment by bittiez** [comment]
Fixes #14

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
     <groupId>me.ryanhamshire</groupId>
     <artifactId>PopulationDensity</artifactId>
-    <version>5.10</version>
+    <version>5.11</version>
```
Comment (RoboMWM) [inline pom.xml:9]: Don't change version in PRs. I'll change it when I make a release.

File: src/me/ryanhamshire/PopulationDensity/PlayerEventHandler.java [hunk 2]
```diff
@@ -37,14 +37,16 @@
...
 	private DataStore dataStore;
 	PopulationDensity instance;
+	private boolean teleportNearbyAnimals;
 
 	// queue of players waiting to join the server
 	public ArrayList<LoginQueueEntry> loginQueue = new ArrayList<LoginQueueEntry>();
 
 	// typical constructor, yawn
-	public PlayerEventHandler(DataStore dataStore, PopulationDensity plugin) {
+	public PlayerEventHandler(DataStore dataStore, PopulationDensity plugin, boolean teleportNearbyAnimals) {
```
Comment (RoboMWM) [inline src/me/ryanhamshire/PopulationDensity/PlayerEventHandler.java:46]: Um... this is not a good idea for any config option.

Either add a public getter in the main method (and access via passed instance) or use a static variable (which is (unfortunately) what the rest of this project does).

File: src/me/ryanhamshire/PopulationDensity/PopulationDensity.java [hunk 3]
```diff
@@ -814,7 +817,7 @@ else if(cmd.getName().equalsIgnoreCase("invite") && player != null)
 			
 			//send a notification to the invitee, if he's available
 			@SuppressWarnings("deprecation")
-            Player invitee = this.getServer().getPlayer(args[0]);			
+            Player invitee = findPlayerByName(args[0]);
```
Comment (RoboMWM) [inline src/me/ryanhamshire/PopulationDensity/PopulationDensity.java:820]: This isn't deprecated anymore. Doesn't matter anyways - the reason it was "deprecated" in the first place was to discourage plugin makers from storing data with player names and to use UUIDs instead.

Please revert this change entirely. You may change this to `getPlayerExact` if you wish

File: src/me/ryanhamshire/PopulationDensity/TeleportPlayerTask.java [hunk 4]
```diff
@@ -15,120 +15,92 @@
...
-import org.bukkit.entity.AnimalTamer;
-import org.bukkit.entity.Animals;
-import org.bukkit.entity.Entity;
-import org.bukkit.entity.EntityType;
-import org.bukkit.entity.LivingEntity;
-import org.bukkit.entity.Ocelot;
-import org.bukkit.entity.Player;
-import org.bukkit.entity.Tameable;
-import org.bukkit.entity.Wolf;
+import org.bukkit.entity.*;
```
Comment (RoboMWM) [inline src/me/ryanhamshire/PopulationDensity/TeleportPlayerTask.java:23]: Please change your IDE settings to _not_ do this. (and try to undo accordingly, thanks).

File: src/me/ryanhamshire/PopulationDensity/TeleportPlayerTask.java [hunk 5]
```diff
@@ -15,120 +15,92 @@
...
+                                        continue;
+                                    }
+                                } else if (type == EntityType.WOLF) {
+                                    Wolf dog = (Wolf) entity;
+                                    if (dog.isSitting()) {
+                                        continue;
+                                    }
+                                }
+
+                                entitiesToTeleport.add(entity);
```
Comment (RoboMWM) [inline src/me/ryanhamshire/PopulationDensity/TeleportPlayerTask.java:76]: What happened to all the newlines (in this entire diff)?

**Review by RoboMWM — CHANGES_REQUESTED** [review]
See comments. The majority of this PR changes the code style, making it inconsistent with the rest of this project.

**Comment by FrankHeijden** [comment]
So... is it fixed? Where can i download the fixed file?

**Comment by bittiez** [comment]
My pull request wasn't accepted, I will try to get you my jar later when I'm not at work, until I can redo this per robo request

**Comment by FrankHeijden** [comment]
ok, thanks :D

**Comment by bittiez** [comment]
http://www.mediafire.com/file/1fo2ap1cqfd7wyd/PopulationDensity.jar

I'll make a new PR when I make the changes you preferred @RoboMWM

**Comment by FrankHeijden** [comment]
Hmm they do still teleport with players :(, first i thought it worked but now are players saying that it isnt anymore

**Comment by FrankHeijden** [comment]
Btw, why is the file of the mediafire link 2.5MB, and the normal plugin just 72KB xD

File: src/me/ryanhamshire/PopulationDensity/PlayerEventHandler.java (hunk 2 above)
Comment (RoboMWM) [reply src/me/ryanhamshire/PopulationDensity/PlayerEventHandler.java:46]: > this is not a good idea

Or rather, this does not follow the conventions of the rest of this plugin afaik

**Comment by RoboMWM** [comment]
> My pull request wasn't accepted,

FYI, I simply requested changes, I didn't just flat-out reject it. Also, you should preferably separate each feature/change in individual PRs instead of lumping them altogether.

> Btw, why is the file of the mediafire link 2.5MB, and the normal plugin just 72KB xD

He probably didn't compile with maven and instead included dependencies or something of that nature.

**Pull request closed without merging**